src/mono/mono/metadata/profiler.c

```diff
@@ -22,6 +22,10 @@ typedef void (*MonoProfilerInitializer) (const char *);
 #define OLD_INITIALIZER_NAME "mono_profiler_startup"
 #define NEW_INITIALIZER_NAME "mono_profiler_init"
 
+#if defined(TARGET_WASM) && defined(MONO_CROSS_COMPILE)
```

Comment:
What is the `MONO_CROSS_COMPILE` for?

Reply:
We want to enable this for the wasm-targeting cross compiler - ie `mono-aot-cross`

Reply:
To explain a bit more (maybe this should go in a comment @pavelsavara).  In order to add tracing instrumentation to AOTed assemblies, the AOT compiler has to be passed a `--profiler=browser` option.  That will cause the AOT compiler to try to load the profiler plugin and the profiler plugin will then tell the AOT compiler to insert instrumentation calls into the AOT images.  


The reason we need to call `init` here is because apparently otherwise `mono_dl_open (NULL, ...)` and `mono_dl_symbol` can't find the browser profiler entrypoint in the statically-linked binary.  (Why not - seems like it should work. 😞 )